File: docs/csharp/scripts/test/testing.nunit.cs
// NUnit — amplamente adotado em projetos enterprise e legados .NET
// Executar: dotnet test
// Convenção: Assert.That(actual, Is.EqualTo(expected)) — actual primeiro

using NUnit.Framework;

static decimal ApplyDiscount(decimal price, int discountPct)
{
    if (discountPct <= 0) return price;
    var discountedPrice = price * (1 - discountPct / 100m);
    return discountedPrice;
}

static string FormatName(string first, string last)
{
    var fullName = $"{first} {last}";
    return fullName;
}

[TestFixture]
public class ApplyDiscountTests
{
    [Test]
    public void AppliesTenPercentDiscountToPrice()
    {
        var price = 100m;

        var actualPrice = ApplyDiscount(price, 10);

        var expectedPrice = 90m;
        Assert.That(actualPrice, Is.EqualTo(expectedPrice));
    }

    [Test]
    public void ReturnsOriginalPriceWhenDiscountIsZero()
    {
        var price = 100m;

        var actualPrice = ApplyDiscount(price, 0);

        var expectedPrice = price;
        Assert.That(actualPrice, Is.EqualTo(expectedPrice));
    }
}

[TestFixture]
public class FormatNameTests
{
    [Test]
    public void FormatsFirstAndLastNameIntoFullName()
    {
        var actualName = FormatName("John", "Doe");

        var expectedName = "John Doe";
        Assert.That(actualName, Is.EqualTo(expectedName));
    }
}

File: docs/vbnet/scripts/test/testing.nunit.vb
' NUnit — framework de testes open-source, popular em projetos .NET Framework
' Executar: dotnet test  ou  Test Explorer no Visual Studio
' Convenção: Assert.That(actual, Is.EqualTo(expected)) — actual primeiro no That()

Imports NUnit.Framework

Function ApplyDiscount(price As Decimal, discountPct As Integer) As Decimal
    If discountPct <= 0 Then Return price

    Dim discountedPrice = price * (1 - discountPct / 100D)

    Return discountedPrice
End Function

Function FormatName(firstName As String, lastName As String) As String
    Dim fullName = $"{firstName} {lastName}"
    Return fullName
End Function

<TestFixture>
Public Class ApplyDiscountTests

    <Test>
    Public Sub AppliesTenPercentDiscountToPrice()
        Dim price = 100D

        Dim actualPrice = ApplyDiscount(price, 10)

        Dim expectedPrice = 90D
        Assert.That(actualPrice, [Is].EqualTo(expectedPrice))
    End Sub

    <Test>
    Public Sub ReturnsOriginalPriceWhenDiscountIsZero()
        Dim price = 100D

        Dim actualPrice = ApplyDiscount(price, 0)

        Dim expectedPrice = price
        Assert.That(actualPrice, [Is].EqualTo(expectedPrice))
    End Sub

End Class

<TestFixture>
Public Class FormatNameTests

    <Test>
    Public Sub FormatsFirstAndLastNameIntoFullName()
        Dim actualName = FormatName("John", "Doe")

        Dim expectedName = "John Doe"
        Assert.That(actualName, [Is].EqualTo(expectedName))
    End Sub

End Class
